services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/postTestTable.tsx

```diff
@@ -0,0 +1,194 @@
+    const { name, completed, id, } = student
+
+    if (!completed) {
+      return <tr key={name}><td className="recommendation-cell empty-cell" /></tr>
+    }
+
+    const isAssigned = previouslyAssignedPostTestStudentIds.includes(id)
+    const isSelected = postTestSelections.includes(id)
+    const isRecommended = studentsWhoCompletedAssignedRecommendations.some(student => student.id === id)
+
+    function toggleSelection() {
+      let newSelections = [...postTestSelections];
+      if (isSelected) {
+        newSelections = newSelections.filter(s => s !== student.id)
+      } else {
+        newSelections.push(student.id);
+      }
+      setPostTestSelections(newSelections)
+    }
+
+    let checkbox = <span aria-label="Unchecked checkbox" className="quill-checkbox unselected" />
+    const assignedText = <div className="assigned">{correctImage}<span>Assigned</span></div>
+
+    if (isSelected) {
+      checkbox = (<span aria-label="Checked checkbox" className="quill-checkbox selected" >
+        <img alt={smallWhiteCheckIcon.alt} src={smallWhiteCheckIcon.src} />
+      </span>)
+    }
+    return (
+      <tr>
+        <td className="recommendation-cell">
+          <button className={`interactive-wrapper ${isRecommended && !isAssigned ? 'recommended' : ''}`} onClick={isAssigned ? () => {} : toggleSelection} type="button">
+            {isAssigned ? assignedText : checkbox}
+            <span />
+          </button>
+        </td>
+      </tr>
+    )
+  }
+
+  const postTestStudentRows = students.map(student => (
+    <PostTestStudentRow
+      key={student.name}
+      postTestSelections={postTestSelections}
+      previouslyAssignedPostTestStudentIds={previouslyAssignedPostTestStudentIds}
+      setPostTestSelections={setPostTestSelections}
+      student={student}
+      studentsWhoCompletedAssignedRecommendations={studentsWhoCompletedAssignedRecommendations}
+    />
+  ))
+
+  const ActivityPackHeader = ({ name, activityPackId, activityCount, handleSelectAllClick, style, }) => {
+    return (
+      <th className="recommendation-header" key={name} style={style || {}}>
+        <div className="name-and-tooltip">
+          <span>{name}</span>
+          <a aria-label="Preview the activity pack" href={`/activities/packs/${activityPackId}`} rel="noopener noreferrer" target="_blank"><img alt="" src={helpIcon.src} /></a>
+        </div>
+        <div className="activity-count-and-select-all">
+          <span className="activity-count">{activityCount} {activityCount === 1 ? 'activity' : 'activities'}</span>
+          <Tooltip
+            tooltipText="Select all column"
+            tooltipTriggerText={<button aria-label="Select all column" className="interactive-wrapper" onClick={handleSelectAllClick} type="button">{ellipsesIcon}</button>}
+          />
+        </div>
+      </th>
+    )
+  }
```

Comment:
I think this is identical to the `ActivityPackHeader` defined in the `recommendationsTable` file, right? If so, it should get split into a third file and imported into both rather than duplicated here.